pkg/postgres/pgutils/retry.go

```diff
@@ -0,0 +1,54 @@
+package pgutils
+
+import (
+	"time"
+
+	"github.com/stackrox/rox/pkg/timeutil"
+)
+
+const (
+	interval = 5 * time.Second
+	timeout  = 5 * time.Minute
+)
+
+// RetryExecQuery is used to specify how long to retry to successfully run an exec query
+// that fails with Transient errors
+func RetryExecQuery(fn func() error) error {
+	// Shape fn to match the RetryQuery below
+	fnWithReturn := func() (struct{}, error) {
+		return struct{}{}, fn()
+	}
+	_, err := RetryQuery(fnWithReturn)
+	return err
+}
+
+// RetryQuery is used to specify how long to retry to successfully run a query
+// that fails with Transient errors
+func RetryQuery[T any](fn func() (T, error)) (T, error) {
+	// Run query immediately
+	if T, err := fn(); err == nil || !isTransientError(err) {
+		return T, err
```

Comment:
```suggestion
	if val, err := fn(); err == nil || !isTransientError(err) {
		return val, err
```

Don't think you intended to use the type param as variable name

Reply:
thx